src/rfid.cpp

```diff
@@ -0,0 +1,191 @@
+#include "rfid.h"
+
+#include "debug.h"
+#include "mqtt.h"
+#include "lcd.h"
+#include "app_config.h"
+#include "RapiSender.h"
+#include "input.h"
+#include "openevse.h"
+#include "Wire.h"
+
+#ifndef I2C_SDA
+#define I2C_SDA 21
+#endif
+
+#ifndef I2C_SCL
+#define I2C_SCL 22
+#endif
+
+RfidTask::RfidTask() :
+  MicroTasks::Task(),
+  _evse(NULL),
+  _scheduler(NULL),
+  _evseStateEvent(this),
+  nfc(PN532_IRQ, PN532_POLLING)
+{
+}
+
+void RfidTask::begin(EvseManager &evse, Scheduler &scheduler){
+    _evse = &evse;
+    _scheduler = &scheduler;
+    MicroTask.startTask(this);
+    // HACK
+    // Schedulers do not work unless this one exists
+    _scheduler->addEvent(100, 0, 0, 0, ~0, EvseState(EvseState::Disabled));
+}
+
+void RfidTask::setup(){
+    _evse->onStateChange(&_evseStateEvent);
+
+    if(!config_rfid_enabled()){
+        status = RFID_STATUS_NOT_ENABLED;
+        return;
+    }
+
+    Wire.begin(I2C_SDA, I2C_SCL);
+    
+    wakeup();
+}
+
+boolean RfidTask::wakeup(){
+    boolean awake = nfc.begin();
+    if(awake){
+        status = RFID_STATUS_ACTIVE;
+    }else{
+        DEBUG.println("RFID module did not respond!");
+        status = RFID_STATUS_NOT_FOUND;
+    }
+    return awake;
+}
+
+void RfidTask::scanCard(){
+    NFCcard = nfc.getInformation();
+    String uidHex = nfc.readUid();
+
+    if(waitingForTag > 0){
+        waitingForTag = 0;
+        cardFound = true;
+        lcd.display("Tag detected!", 0, 0, 0, LCD_CLEAR_LINE);
+        lcd.display(uidHex, 0, 1, 3000, LCD_CLEAR_LINE);
+    }else{
+        // Check if tag is stored locally
+        char storedTags[rfid_storage.length() + 1];
+        rfid_storage.toCharArray(storedTags, rfid_storage.length()+1);
+        char* storedTag = strtok(storedTags, ",");
+        while(storedTag)
+        {
+            String storedTagStr = storedTag;
+            storedTagStr.trim();
+            uidHex.trim();
+            if(storedTagStr.compareTo(uidHex) == 0){
+                rapiSender.sendCmd(F("$FE"));
+                break;
+            }
+            storedTag = strtok(NULL, ",");
+        }
+
+        // Send to MQTT broker
+        DynamicJsonDocument data(4096);
+        data["rfid"] = uidHex;
+        mqtt_publish(data);
+    }
+}
+
+void RfidTask::startTimer(uint8_t seconds){
+    static int day = 0;
+    static int32_t offset = 0;
+    _scheduler->getCurrentTime(day, offset);
+    offset += seconds;
+    int schedulerDay = 1 << day;
+    int hour = offset / 3600;
+    int minute = (offset % 3600) / 60;
+    int second = offset % 60;
+    _scheduler->addEvent(0xFFFFFFFF, hour, minute, second, schedulerDay, EvseState(EvseState::Disabled));
+}
+
+void RfidTask::abortTimer(){
+    _scheduler->removeEvent(0xFFFFFFFF);
+}
```

Comment:
This does not seem to stop the event from getting triggered. Is this to be expected? How do I abort an event in the scheduler that has not yet been triggered?